v1/daemon/image_test.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2018 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package daemon
+
+import (
+	"context"
+	"io"
+	"os"
+	"reflect"
+	"testing"
+
+	"github.com/google/go-containerregistry/name"
+
+	"github.com/google/go-containerregistry/v1/tarball"
+)
+
+var imagePath = "../tarball/test_image_1.tar"
+
+type MockImageSaver struct {
+	path string
+}
+
+func (m *MockImageSaver) ImageSave(_ context.Context, _ []string) (io.ReadCloser, error) {
+	return os.Open(m.path)
+}
+
+func TestImage(t *testing.T) {
+	getImageSaver = func() (ImageSaver, error) {
+		return &MockImageSaver{path: imagePath}, nil
+	}
```

Comment:
Do you need/want to reset `getImageSaver` after this test completes? Or if this is the `getImageSaver` that should be used for all tests, consider putting it in `func init()` ?